### ISSUE TITLE
wpt editor: Indicating FP entries

### ISSUE THREAD

**Issue by michihdeu** [TravelMapping/Web#396]
Once we can load the wpt file from Git, see #372 the next step would be indicating
- FP entries of the route, see #375 (from DB or directly calculated in wpt editor and updated with every change? Button to "recalculate")
- [url=http://travelmapping.net/logs/unmatchedfps.log]Unmatched FPs[/url] of the route
- [url=http://travelmapping.net/logs/nearmatchfps.log]Near match FPs[/url] of the route

**Comment by yakra** [comment]
Not contingent on loading files from GitHub.
Is this not just a duplicate of #375?

**Comment by michihdeu** [comment]
The first one is a duplicate ("Add capability to click on an entry in the Err.? column for a dialog listing its FP code.")
The second and third one not.

#375 could be a _quick_ implementation (step 1) without any redesign of the wpt editor.
#396 is the presentation of new, existing and nonexisting FP entries in the wpt editor (anywhere down the road).

**Comment by yakra** [comment]
Took me a little bit to see the sense in this.
* My 1st thought was *"we need to edit nearmatchfps.csv, completely different from the .wpt file we're using wptedit to edit"*, and this info would just be presented in the rare case someone's editing the file for some reason anyway.
* But as @michihdeu noted in the OP, #372. This makes sense if someone is loading the file into wptedit via a link from the datacheck page.

Maybe have links in the info box; click to produce a dialog box...
> 12 waypoints: 10 visible, 2 hidden.
> `1 waypoints with possible errors.`
> **eng.b1113 has [1 unmatched FP](https://github.com/), with [1 near match FP](https://github.com/)**
> Length: 16.97 mi, 27.31 km
> Average spacing: 1.54 mi, 2.48 km
> Average visible spacing: 1.89 mi, 3.03 km

We'd need new DB tables for unmatchedfps and nearmatchfps.
The queries would be pretty fast & painless though.

Complications:
* What to do if the state of the file being edited changes so that an unmatched FP entry would be matched again?
  - Just don't list that entry (this looks more sensible)?
  - List it with a note?
* Similarly, watch the state of existing Marked FPs, and see if they'd become unmatched, and need to be removed or changed.

**Comment by yakra** [comment]
Related:
* Check for Marked FPs, and indicate these in a different color in the WP table, TDB.
  - If a WP has 2+ errors with some-but-not-all marked FP? I guess maybe use the more "urgent" color, the existing lightsalmon-ish?
* `2 waypoints with possible errors.` could become `2 waypoints with possible errors, 1 not marked FP` or somesuch.

**Comment by michihdeu** [comment]
> If a WP has 2+ errors with some-but-not-all marked FP? I guess maybe use the more "urgent" color, the existing lightsalmon-ish?

yep, the more "urgent" color.

**Comment by yakra** [comment]
I wrote:
> We'd need new DB tables for unmatchedfps and nearmatchfps.

Wait, what? :P

**Comment by michihdeu** [comment]
Having the FP entries would be great - especially the SA errors.
The rest is _nice to have_.
